packages/shared/src/components/drawers/NavDrawerItem.tsx

```diff
@@ -3,11 +3,10 @@ import Link from 'next/link';
 import { Button, ButtonProps, ButtonVariant } from '../buttons/Button';
 import ConditionalWrapper from '../ConditionalWrapper';
 
-export interface NavItemProps
-  extends Pick<ButtonProps<'a'>, 'href' | 'icon' | 'onClick'> {
+export type NavItemProps = ButtonProps<'a'> & {
```

Comment:
Why did we end up using `type` on this over `interface`?

Reply:
`ButtonProps` is a type and this is the only way to extend it properly. Otherwise we would need to use the `Pick` thingy with only a handful of listed props. Doing it this way, we can pass through any prop that the Button supports.